subprojects/ide/src/test/groovy/org/gradle/plugins/ide/internal/tooling/eclipse/EclipseModelBuilderTest.groovy

```diff
@@ -66,6 +67,20 @@ class EclipseModelBuilderTest extends AbstractProjectBuilderSpec {
         eclipseModel.projectNatures.collect { it.id } == ['nature.a', 'nature.b']
     }
 
+    def "can modify natures before and after merging"() {
+        setup:
+        project.eclipse.project.natures = ['nature.a']
+        project.eclipse.project.file.beforeMerged { it.natures += 'nature.b' }
+        project.eclipse.project.file.whenMerged { it.natures += 'nature.c' }
+        def modelBuilder = createEclipseModelBuilder()
+
+        when:
+        def eclipseModel = modelBuilder.buildAll("org.gradle.tooling.model.eclipse.EclipseProject", project)
+
+        then:
+        eclipseModel.projectNatures.collect { it.id } == ['nature.b', 'nature.a', 'nature.c']
+    }
+
```

Comment:
❌This change can break the IDE setup if somebody first used `eclipse` Gradle plugin then migrated to Buildship and left some code in`file.beforeMerged`/`file.whenMerged`. I'd suggest explaining the changes in the [release notes](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/release/notes.md) as a potential breaking change.